### PR TITLE
Feature/em 1386 arrange bookmarks

### PR DESCRIPTION
**Pull request by benouaer**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EM-1386

### Change description ###
enable arranging of bookmarks with drag and drop and nesting

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: projects/media-viewer/src/assets/sass/angular-tree-component.scss [hunk 1]
```diff
@@ -0,0 +1,84 @@
+.tree-children.tree-children-no-padding { padding-left: 0 }
+.tree-children { padding-left: 20px; overflow: hidden }
```
Comment (fazanki) [inline projects/media-viewer/src/assets/sass/angular-tree-component.scss:2]: not sure if worth  turning this into the scss style rather then css

File: projects/media-viewer/src/lib/viewers/pdf-viewer/side-bar/bookmarks/bookmarks.component.ts [hunk 2]
```diff
@@ -52,6 +54,46 @@ export class BookmarksComponent implements OnInit, OnDestroy {
     this.editableBookmark = id;
   }
 
+  onBookmarkMove({ node, from, to }: any) {
+    let movedBookmarks = [];
+    let fromNext = this.getSibling(from, from.index);
+    fromNext = fromNext && fromNext.id === node.previous ? this.getSibling(from, from.index + 1) : fromNext;
+    if (fromNext) {
+      movedBookmarks.push({ ...fromNext, previous: node.previous })
```
[the remark this answers is not in the document]
Comment (benouaer) [reply projects/media-viewer/src/lib/viewers/pdf-viewer/side-bar/bookmarks/bookmarks.component.ts:62]: agreed

File: projects/media-viewer/src/lib/media-viewer.module.ts [hunk 3]
```diff
@@ -33,8 +33,10 @@ import { IcpDirective } from './icp/icp.directive';
 import { IcpPresenterService } from './icp/icp-presenter.service';
 import { IcpFollowerService } from './icp/icp-follower.service';
 import { ConfirmActionDialogComponent } from './icp/confirm-exit/confirm-action-dialog.component';
+import { TreeModule } from 'angular-tree-component';
 
 
+// noinspection AngularInvalidImportedOrDeclaredSymbol
```
[the remark this answers is not in the document]
Comment (benouaer) [reply projects/media-viewer/src/lib/media-viewer.module.ts:39]: not sure how it ended up there, but I'll remove it

File: projects/media-viewer/src/lib/store/effects/bookmarks.effects.ts [hunk 4]
```diff
@@ -41,14 +41,32 @@ export class BookmarksEffects {
...
     map((action: bookmarksActions.DeleteBookmark) => action.payload),
-    exhaustMap((bookmarkId) =>
-      this.bookmarksApiService.deleteBookmark(bookmarkId)
+    exhaustMap(({ deleted, updated }) =>
+      this.bookmarksApiService.deleteMultipleBookmarks({ deleted, updated })
         .pipe(
-          map(() => new bookmarksActions.DeleteBookmarkSuccess(bookmarkId)),
+          switchMap(() => {
+            const success: any[] = [new bookmarksActions.DeleteBookmarkSuccess(deleted)];
+            if (updated) {
```
[the remark this answers is not in the document]
Comment (benouaer) [reply projects/media-viewer/src/lib/store/effects/bookmarks.effects.ts:65]: agreed

File: projects/media-viewer/src/assets/sass/angular-tree-component.scss (hunk 1 above)
Comment (benouaer) [reply projects/media-viewer/src/assets/sass/angular-tree-component.scss:2]: there's quite a few of them, and they belong to the external library, so I'm inclined not to

**Pull request merged**